src/ape_ethereum/transactions.py

```diff
@@ -156,6 +159,54 @@ def raise_for_status(self):
             txn_hash = HexBytes(self.txn_hash).hex()
             raise TransactionError(message=f"Transaction '{txn_hash}' failed.")
 
+    def decode_logs(
+        self, abi: Optional[Union[EventABI, ContractEvent]] = None
+    ) -> Iterator[ContractLog]:
+        if not abi:
+            # Check for DS-Note library logs.
+            for log in self.logs:
+                try:
+                    yield self._decode_ds_note(log)
+                except DecodingError:
+                    continue
+
+        return super().decode_logs(abi)
```

Comment:
So this will add two loops worth of processing to this method, is there a better way of folding this in, or a heuristic that can be used to reduce the amount of processing? Picturing a transaction w/ several thousand logs it might be slow-ish